docs/src/tutorial/embedding.rst

```diff
@@ -58,30 +57,123 @@ to export it as a linker symbol that can be used by other C files, which in this
 case is ``embedded_main.c``.
 
 .. literalinclude:: ../../examples/tutorial/embedding/embedded.pyx
+   :lines: 4-
+   :caption: embedded.pyx
 
 The C ``main()`` function of your program could look like this:
 
 .. literalinclude:: ../../examples/tutorial/embedding/embedded_main.c
+    :caption: embedded_main.c
     :linenos:
     :language: c
 
 (Adapted from the `CPython documentation
 <https://docs.python.org/3/extending/extending.html#the-module-s-method-table-and-initialization-function>`_.)
 
 Instead of writing such a ``main()`` function yourself, you can also let
-Cython generate one into your module's C file with the ``cython --embed``
-option.  Or use the
-`cython_freeze <https://github.com/cython/cython/blob/master/bin/cython_freeze>`_
-script to embed multiple modules.  See the
-`embedding demo program <https://github.com/cython/cython/tree/master/Demos/embed>`_
-for a complete example setup.
-
-Be aware that your application will not contain any external dependencies that
-you use (including Python standard library modules) and so may not be truly portable.
-If you want to generate a portable application we recommend using a specialized
-tool (e.g. `PyInstaller <https://pyinstaller.org/en/stable/>`_
-or `cx_freeze <https://cx-freeze.readthedocs.io/en/latest/index.html>`_) to find and
-bundle these dependencies.
+Cython generate one into your module's C file with the ``cython --embed`` option.
+
+When using ``--embed``, Cython generates the C implementation file (e.g., ``embedded.c``) along with a header file (e.g., ``embedded.h``).
+
+The generated ``embedded.c`` file contains a ``__Pyx_main()`` function. This function initializes the Python interpreter and can be modified to call public functions exported by your Cython module (such as ``say_hello_from_python()``).
```

Comment:
Let's break overly long lines. Especially a new sentence is usually a good place to start a new line.
```suggestion
When using ``--embed``, Cython generates the C implementation file (e.g., ``embedded.c``)
along with a header file (e.g., ``embedded.h``).

The generated ``embedded.c`` file contains a ``__Pyx_main()`` function.
This function initializes the Python interpreter and can be modified to call public functions
exported by your Cython module (such as ``say_hello_from_python()``).
```